Booktrix/Controllers/SignStoryboard/SignInViewModel.swift

```diff
@@ -0,0 +1,57 @@
+//
+//  SignInViewModel.swift
+//  Booktrix
+//
+//  Created by Impresyjna on 14.05.2017.
+//  Copyright © 2017 Impresyjna. All rights reserved.
+//
+import UIKit
+
+struct SignInForm {
+    var login: String
+    var password: String
+}
+
+final class SignInViewModel {
+    var form: SignInForm
+    
+    
+    typealias SignInCompletion = (ApiResponse<User>) -> ()
+    
+    init() {
+        form = SignInForm(login: "",  password: "")
+    }
+    
+    func login(completion: @escaping SignInCompletion) {
+        let service = UserService()
+        
+        let status = FieldValidationName(name: LocalizedString.login, validationResult: form.login.validate(.nonEmpty)).result.combine([
+            FieldValidationName(name: LocalizedString.password, validationResult: form.password.validate(.nonEmpty, .atLeast(6))).result
+            ])
+        
+        switch status {
+        case .success:
+            service.login(with: form, completion: { (result) in
+                switch result {
+                case .success(let user):
+                    KeychainStorage().setUser(user)
+                    completion(.success(user))
+                case .codeFailure(var error):
+                    error.message = LocalizedString.badAuth
```

Comment:
Zrobiłbym porządek z tymi errorami. Jeśli chcesz dostawać error 500 to nie twórz go sztucznie tylko pozwól by api client go zwrócił (czyli zmień walidację).